File: android/app/src/main/java/com/example/app/domain/navcaddy/models/IntentType.kt
package caddypro.domain.navcaddy.models

/**
 * Enum representing all supported MVP intent types in the NavCaddy Engine.
 *
 * These intents map user natural language inputs to structured routing targets
 * across the app's modules (Caddy, Coach, Recovery, Settings).
 *
 * Spec reference: navcaddy-engine.md R2, Q1
 * Live Caddy reference: live-caddy-mode.md R5, live-caddy-mode-plan.md Task 24
 */
enum class IntentType {
    /** Request to adjust club distances or yardage expectations */
    CLUB_ADJUSTMENT,

    /** Query about current recovery status or readiness */
    RECOVERY_CHECK,

    /** Request for shot recommendation (club, strategy, aim) */
    SHOT_RECOMMENDATION,

    /** User wants to enter or update score for a hole */
    SCORE_ENTRY,

    /** Query about historical miss patterns or tendencies */
    PATTERN_QUERY,

    /** Request for a specific drill or practice suggestion */
    DRILL_REQUEST,

    /** Query about current or forecast weather conditions */
    WEATHER_CHECK,

    /** Request for statistics (scoring, club performance, etc.) */
    STATS_LOOKUP,

    /** User is starting a new round */
    ROUND_START,

    /** User is ending the current round */
    ROUND_END,

    /** Query about equipment (club specs, bag contents) */
    EQUIPMENT_INFO,

    /** Query about course information (hole details, layout) */
    COURSE_INFO,

    /** Request to change app settings or preferences */
    SETTINGS_CHANGE,

    /** User needs help or instructions */
    HELP_REQUEST,

    /** User providing feedback about the app */
    FEEDBACK,

    // Live Caddy Mode intents (Task 24)

    /** Query about bailout location or safe miss area */
    BAILOUT_QUERY,

    /** Query about readiness score and how it affects strategy in Live Caddy mode */
    READINESS_CHECK
}

File: ios/App/Sources/Core/Domain/NavCaddy/IntentType.swift
import Foundation

/// All supported intent types in the NavCaddy Engine MVP.
///
/// Maps to the 15 MVP intents defined in spec R2.
enum IntentType: String, Codable, CaseIterable, Hashable {
    // Core intents
    case clubAdjustment = "club_adjustment"
    case recoveryCheck = "recovery_check"
    case shotRecommendation = "shot_recommendation"
    case scoreEntry = "score_entry"
    case patternQuery = "pattern_query"

    // Extended intents
    case drillRequest = "drill_request"
    case weatherCheck = "weather_check"
    case statsLookup = "stats_lookup"
    case roundStart = "round_start"
    case roundEnd = "round_end"

    // Additional intents
    case equipmentInfo = "equipment_info"
    case courseInfo = "course_info"
    case settingsChange = "settings_change"
    case helpRequest = "help_request"
    case feedback = "feedback"
}
